spec/support/media_spec_helpers.rb

```diff
@@ -0,0 +1,65 @@
+module MediaSpecHelpers
+  module FileHandling
+    def media_fixture(name)
+      path = File.expand_path("../../fixtures/media/#{name}", __FILE__)
+      File.open(path)
+    end
+  end
+
+  shared_context 'media helpers' do
+    def factory_name(described_class)
+      described_class.name.underscore.gsub('/', '_').to_sym
+    end
+
+    def file_for_type(file_type)
```

Comment:
Ah ok nevermind. Looks great.

On Thu, Jan 21, 2016 at 3:07 PM, Melody <notifications@github.com> wrote:

> In spec/support/media_spec_helpers.rb
> <https://github.com/thecartercenter/elmo/pull/175#discussion_r50455761>:
>
> > @@ -0,0 +1,65 @@
> > +module MediaSpecHelpers
> > +  module FileHandling
> > +    def media_fixture(name)
> > +      path = File.expand_path("../../fixtures/media/#{name}", __FILE__)
> > +      File.open(path)
> > +    end
> > +  end
> > +
> > +  shared_context 'media helpers' do
> > +    def factory_name(described_class)
> > +      described_class.name.underscore.gsub('/', '_').to_sym
> > +    end
> > +
> > +    def file_for_type(file_type)
>
> since they have different file names & directories, i'd either need to
> pass the file path into the shared example, or move that same case logic
> into the example group definition.
>
> The reason this works differently from the other specs is because I can't
> leverage the factories as much for it, with traits I can create a
> :media_audio with an mp3 given only the type & the extension, but I can't
> create a :media_video with a png without setting the item directly, since
> the png trait is only valid & defined for the :media_image factory. I
> could make all of the traits completely generic to all media objects, in
> which case I'd be able to handle this based on extensions alone, but that
> makes the webm case weird since it's valid for multiple content types,
> and seems like a worse practice in general, since it loses the structure &
> most of those combinations wouldn't be valid.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/thecartercenter/elmo/pull/175/files#r50455761>.
>



-- 
Tom Smyth

Worker-Owner, Sassafras Tech Collective
Specializing in innovative, usable tech for social change
sassafras.coop *·* @sassafrastech

Resident, Touchstone Cohousing
touchstonecohousing.org